src/machine-learning/machine-learning.ts

```diff
@@ -187,11 +184,10 @@ export class MachineLearning implements FirebaseServiceInterface {
           throw new FirebaseMachineLearningError(
             'internal-error',
             `Error during signing upload url: ${err.message}`);
-
-        });
+        }) as Promise<ModelContent>;
```

Comment:
This is kind of weird. Just define your constant to be `ModelContent`.

```
const modelContent = deepCopy(options) as ModelContent;
```

Reply:
If I do that I wouldn't be able to assign to it here: modelContent.tfliteModel!.gcsTfliteUri = uri;
because all the ModelContent properties are read only.
(That's why I initially left it as Promise<object>)